test/message_bus/message_inbox/inbox_message_hash.js

```diff
@@ -0,0 +1,143 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+const web3 = require('../../test_lib/web3.js');
+const Utils = require('../../test_lib/utils.js');
+const { AccountProvider } = require('../../test_lib/utils.js');
+const MessageInbox = artifacts.require('MessageInboxDouble');
+
+const BN = require('bn.js');
+
+contract('MessageInbox::inboxMessageHash', (accounts) => {
+  let messageInbox;
+  const setupParams = {};
+  const accountProvider = new AccountProvider(accounts);
+  beforeEach(async () => {
+    messageInbox = await MessageInbox.new();
+    setupParams.intentHash = Utils.getRandomHash();
+    setupParams.nonce = new BN(Utils.getRandomNumber(63)); 
+    setupParams.feeGasPrice = new BN(Utils.getRandomNumber(9999));
+    setupParams.feeGasLimit = new BN(Utils.getRandomNumber(10000));
+    setupParams.sender = messageInbox.address;
+    setupParams.messageOutbox = await accountProvider.get();
+    setupParams.metachainId = Utils.generateRandomMetachainId();
+    setupParams.outboxStorageIndex = new BN(Utils.getRandomNumber(63));
+    setupParams.stateRoot = await accountProvider.get();
+    setupParams.maxStorageRootItems = new BN(Utils.getRandomNumber(500));
+  });
+
+  contract('Positive Tests', async () => {
+    it('Should successfully generate inbox message hash', async () => {
+      await messageInbox.setupMessageInboxExternal(
+        setupParams.metachainId,
+        setupParams.messageOutbox,
+        setupParams.outboxStorageIndex,
+        setupParams.stateRoot,
+        setupParams.maxStorageRootItems,
+      );
+
+      const messageHash = await messageInbox.inboxMessageHash(
+        setupParams.intentHash,
+        setupParams.nonce,
+        setupParams.feeGasPrice,
+        setupParams.feeGasLimit,
+        setupParams.sender,
+      );
+
+      const MESSAGE_TYPEHASH = web3.utils.keccak256(
+        'Message(bytes32 intentHash,uint256 nonce,uint256 feeGasPrice,uint256 feeGasLimit,address sender)',
+      );
+
+      const MMB_CHANNEL_TYPEHASH = web3.utils.keccak256(
+        'MosaicMessageBusChannel(address outbox, address inbox)',
+      );
+
+      const MMB_DOMAIN_SEPARATOR_TYPEHASH = web3.utils.keccak256(
+        'MosaicMessageBus(string name,string version,bytes32 metachainId,bytes32 channelSeparator)',
+      );
+
+      const MMB_DOMAIN_SEPARATOR_NAME = 'Mosaic-Bus';
+      const MMB_DOMAIN_SEPARATOR_VERSION = '0';
+
+      const channelSeparator = web3.utils.keccak256(
+        web3.eth.abi.encodeParameters(
+          [
+            'bytes32',
+            'address',
+            'address',
+          ],
+          [
+            MMB_CHANNEL_TYPEHASH,
+            setupParams.messageOutbox,
+            messageInbox.address,
+          ],
+        ),
+      );
+
+      const expectedinboundChannelIdentifier = web3.utils.keccak256(
```

Comment:
This logic is already implemented in test/consensus-gateway/utils.js.
You can reuse it and avoid duplication here.

Reply:
Used test/consensus-gateway/utils.js file to implement the logic.